### PR TITLE
Update inet_aton.c from FreeBSD uptream removes advertising clause

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: 

One-line summary:
The new https://github.com/freebsd/freebsd/blob/455de2f90f71a495eb5068b060c17de9bfcffda6/lib/libc/inet/inet_addr.c **removes advertising clause** from license.

Additional description (if needed):
The only user of inet_aton() within eggdrop is one function in dns module. I tested this function and enhanced its logging. But you cant see the `DNS: Valid dns-server` message, because #948 is not merged yet. egg_inet_aton() now uses isdigit(), islower(), isspace() instead of egg_*() due to all of them being posix 2001.
Removal of the advertising clause helps with #226 see https://github.com/eggheads/eggdrop/issues/226#issuecomment-295974628_

Test cases demonstrating functionality (if applicable):
```
$ grep INET_ATON config.h
#define HAVE_INET_ATON 1
$ nm eggdrop|grep aton
                 U inet_aton@@GLIBC_2.2.5

$ grep INET_ATON config.h
/* #define HAVE_INET_ATON 1 */
$ nm eggdrop|grep aton
00000000000589c0 T egg_inet_aton
```

### CODE REVIEW

File: src/compat/inet_aton.c [hunk 1]
```diff
@@ -99,19 +102,14 @@ static char rcsid[] = "$-Id: inet_addr.c,v 1.11 1999/04/29 18:19:53 drepper Exp
...
  */
-int egg_inet_aton(cp, addr)
-const char *cp;
-struct in_addr *addr;
-{
-  static const uint32_t max[4] = { 0xffffffff, 0xffffff, 0xffff, 0xff };
-  uint32_t val;       /* changed from u_long --david */
-  int base;
-  int n;
+int
```
Comment (vanosg) [inline src/compat/inet_aton.c]: multi-line

**Review by vanosg — COMMENTED** [review]
Please test the cases that are affected by these changes and show code equivalency

**Comment by vanosg** [comment]
This is more than just removing the advertisement, there are some semi-significant updates to the code being run. This would require some extensive testing to validate- please provide test cases for the new code in use. Thanks!

File: src/compat/inet_aton.c (hunk 1 above)
Comment (michaelortmann) [reply src/compat/inet_aton.c]: done

**Comment by michaelortmann** [comment]
vanosg: As we spoke, the multi line fix is committed and thus this PR is ready for review again.

**Pull request merged**